File: sams-backend/src/main/java/com/sams/entity/MailNotification.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_mail_notification", schema = "base")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MailNotification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "mail_notification_id")
    private Long mailNotificationId;

    @Column(name = "main_process_id")
    private Long mainProcessId;

    @Column(name = "sub_process_id")
    private Long subProcessId;

    @Column(name = "sub_process_name")
    private String subProcessName;

    @Column(name = "main_process_name")
    private String mainProcessName;

    @Column(name = "to_email")
    private String toEmail;

    @Column(name = "cc_email")
    private String ccEmail;

    @Column(name = "bcc_email")
    private String bccEmail;

    @Column(name = "org_id")
    private Long orgId;

    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "active")
    private String active;

    @Column(name = "active_disp")
    private Boolean activeDisp;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt_disp")
    private String createdDtDisp;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt_disp")
    private String updatedDtDisp;

    @Column(name = "org_name")
    private String orgName;

    @Column(name = "location_name")
    private String locationName;

    @PrePersist
    protected void onCreate() {
       // Auto-generated
    }

    @PreUpdate
    protected void onUpdate() {
       // Auto-generated
    }
}

File: SAMSUI/src/app/Model/master/mailNotification.ts
import { BaseModel } from "../base/baseModel";

export class MailNotificationModel extends BaseModel {

    public mailNotificationId ?: number;

    public mainProcessId ?: number;

    public subProcessId ?: number;

    public subProcessName ?: string;

    public  mainProcessName?: string;

    public toEmail ?: string;

    public ccEmail ?: string;

    public bccEmail ?: string;

    public orgId ?: number;

    public locationId ?: number;

    public active ?: string;

    public activeDisp ?:boolean;

    public createdBy ?:string;

    public createdDtDisp ?: string;

    public updatedBy ?: string;

    public updatedDtDisp ?: string;

    public pageNumber ?: number;

    public recordsPerPage ?: number;

    public orgName ?: string;

    public locationName ?: string;
}
